### PR TITLE
Refactor/recipe recipe repo class

### PR DESCRIPTION
**Pull request by michannstoner**
# Pull Request Template
## Description
- Refactor properties and methods on Recipe class, as well as it's accompanying test. Also refactors methods for RecipeRepo class, and accompanying tests. 

## Type of change
Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/recipe.js [hunk 1]
```diff
@@ -1,26 +1,45 @@
...
+  constructor(name, id, image, ingredients, instructions, tags) {
+    this.name = name;
+    this.id = id;
+    this.image = image;
+    this.ingredients = ingredients;
+    this.instructions = instructions;
+    this.tags = tags;
+  }
+
+  getIngredientNames() {
```
Comment (bhohnco) [inline src/recipe.js:13]: Nice work on this function, looks solid!

File: src/recipeRepository.js [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
+    this.recipes.filter(recipe => {
+      if (recipe.tags.includes(recipeTag)) {
+        taggedRecipes.push(recipe);
+      }
+    });
+    return taggedRecipes;
+  }
+
+  findRecipe(searchTerm) {
+    const searchResults = this.recipes.reduce((results, recipe) => {
```
Comment (bhohnco) [inline src/recipeRepository.js:17]: Nice use of reduce here

**Review by bhohnco — APPROVED** [review]

File: test/recipeRepository-test.js [hunk 3]
```diff
@@ -1,27 +1,278 @@
...
+  it('should hold recipe data', () => {
+    
+    expect(recipeRepository.recipes).to.equal(recipeTestData);
   });
 
-  describe('findRecipe', () => {
-    it('Should be able to filter through its array by ingredients', () => {
-      expect(recipeRepository.findRecipe('yolk').length).to.equal(2);
-    });
+  it('should return recipes by tags', () => {
```
Comment (RMartin0717) [inline test/recipeRepository-test.js:264]: There's a similar method in the User class that we might want to look at potentially creating a helper method for later if it's possible to reuse it. With that one, it's for filtering by tag in the User favorite recipes. This is more a note to possibly revisit that later.

File: test/recipe-test.js [hunk 4]
```diff
@@ -1,32 +1,127 @@
...
+      "Add egg and vanilla and mix until combined."
+    ]);
+  });
 
-});
+  it('should calculate the cost of its ingredients', () => {
+      
+    expect(recipe.calculateCost()).to.equal(5.04);
+  });
+});
```
Comment (RMartin0717) [inline test/recipe-test.js:127]: Tests look great! Good job making sure everything got covered that wasn't being tested before

**Pull request merged**